subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -0,0 +1,838 @@
+
+	return err;
+}
+#else
+static int broker_init(void)
+{
+	int err;
+	struct addrinfo *result;
+	struct addrinfo *addr;
+	struct addrinfo hints = {
+		.ai_family = AWS_AF_FAMILY,
+		.ai_socktype = SOCK_STREAM
+	};
+
+	err = getaddrinfo(CONFIG_AWS_IOT_BROKER_HOST_NAME,
+			  NULL, &hints, &result);
+	if (err) {
+		LOG_ERR("getaddrinfo, error %d", err);
+		return err;
+	}
+
+	addr = result;
+
+	while (addr != NULL) {
+		if ((addr->ai_addrlen == sizeof(struct sockaddr_in)) &&
+		    (AWS_AF_FAMILY == AF_INET)) {
+			struct sockaddr_in *broker4 =
+				((struct sockaddr_in *)&broker);
+			char ipv4_addr[NET_IPV4_ADDR_LEN];
+
+			broker4->sin_addr.s_addr =
+				((struct sockaddr_in *)addr->ai_addr)
+				->sin_addr.s_addr;
+			broker4->sin_family = AF_INET;
+			broker4->sin_port = htons(CONFIG_AWS_IOT_PORT);
+
+			inet_ntop(AF_INET, &broker4->sin_addr.s_addr, ipv4_addr,
+				  sizeof(ipv4_addr));
+			LOG_DBG("IPv4 Address found %s", log_strdup(ipv4_addr));
+			break;
+		} else if ((addr->ai_addrlen == sizeof(struct sockaddr_in6)) &&
+			   (AWS_AF_FAMILY == AF_INET6)) {
+			struct sockaddr_in6 *broker6 =
+				((struct sockaddr_in6 *)&broker);
+			char ipv6_addr[NET_IPV6_ADDR_LEN];
+
+			memcpy(broker6->sin6_addr.s6_addr,
+			       ((struct sockaddr_in6 *)addr->ai_addr)
+			       ->sin6_addr.s6_addr,
+			       sizeof(struct in6_addr));
+			broker6->sin6_family = AF_INET6;
+			broker6->sin6_port = htons(CONFIG_AWS_IOT_PORT);
+
+			inet_ntop(AF_INET6, &broker6->sin6_addr.s6_addr,
+				  ipv6_addr, sizeof(ipv6_addr));
+			LOG_DBG("IPv4 Address found %s", log_strdup(ipv6_addr));
+			break;
+		}
+
+		LOG_DBG("ai_addrlen = %u should be %u or %u",
+			(unsigned int)addr->ai_addrlen,
+			(unsigned int)sizeof(struct sockaddr_in),
+			(unsigned int)sizeof(struct sockaddr_in6));
+
+		addr = addr->ai_next;
+		break;
+	}
+
+	freeaddrinfo(result);
+
+	return err;
+}
+#endif
+
+static int client_broker_init(struct mqtt_client *const client)
+{
+	int err;
+
+	mqtt_client_init(client);
+
+	err = broker_init();
+	if (err) {
+		return err;
+	}
+
+	client->broker			= &broker;
+	client->evt_cb			= mqtt_evt_handler;
+	client->client_id.utf8		= (char *)client_id_buf;
+	client->client_id.size		= strlen(client_id_buf);
+	client->password		= NULL;
+	client->user_name		= NULL;
+	client->protocol_version	= MQTT_VERSION_3_1_1;
+	client->rx_buf			= rx_buffer;
+	client->rx_buf_size		= sizeof(rx_buffer);
+	client->tx_buf			= tx_buffer;
+	client->tx_buf_size		= sizeof(tx_buffer);
+	client->transport.type		= MQTT_TRANSPORT_SECURE;
+
+	static sec_tag_t sec_tag_list[] = { CONFIG_AWS_IOT_SEC_TAG };
+	struct mqtt_sec_config *tls_cfg = &(client->transport).tls.config;
+
+	tls_cfg->peer_verify		= 2;
+	tls_cfg->cipher_count		= 0;
+	tls_cfg->cipher_list		= NULL;
+	tls_cfg->sec_tag_count		= ARRAY_SIZE(sec_tag_list);
+	tls_cfg->sec_tag_list		= sec_tag_list;
+	tls_cfg->hostname		= CONFIG_AWS_IOT_BROKER_HOST_NAME;
+
+	return err;
+}
+
+int aws_iot_ping(void)
+{
+	return mqtt_live(&client);
```

Comment:
Let's use `mqtt_ping()` here instead, to avoid having to do workarounds to not go to much paste the kee-alive

Reply:
Agreed. updated